lib/solve/dependency.rb

```diff
@@ -27,12 +27,14 @@ def initialize(artifact, name, constraint = Semverse::DEFAULT_CONSTRAINT)
     def to_s
       "#{name} (#{constraint})"
     end
+    alias :inspect :to_s
 
     # @param [Object] other
     #
     # @return [Boolean]
     def ==(other)
       other.is_a?(self.class) &&
+        name == other.name &&
```

Comment:
Previously, the solver was checking to ensure that we'd not seen the requirement to use a dependency before. ( https://github.com/CocoaPods/Molinillo/blob/master/lib/molinillo/resolution.rb#L754 ) . We'd compare two dependencies, and if they both had the same *parent* - the artifact - and the same constraint, they'd pass, regardless of whether the *name* of the dependency was the same.